pennylane/ops/functions/bind_new_parameters.py

```diff
@@ -198,8 +198,11 @@ def bind_new_parameters_pow(op: Pow, params: Sequence[TensorLike]):
     return Pow(bind_new_parameters(op.base, params), op.scalar)
 
 
-@bind_new_parameters.register
-def bind_new_parameters_hamiltonian(op: qml.Hamiltonian, params: Sequence[TensorLike]):
```

Comment:
We should have a different kernel for this, since a `LinearCombination`'s data might contain non-coeffs.

Reply:
```
>>> H = 2.0 * qml.Hermitian(np.eye(2), 0) + 3 * qml.X(0)
>>> H
(
    2.0 * Hermitian(array([[1., 0.],
       [0., 1.]]), wires=[0])
  + 3 * X(0)
)
>>> H.data
(2.0,
 array([[1., 0.],
        [0., 1.]]),
 3)
>>> qml.ops.functions.bind_new_parameters(H, (1.0, -np.eye(2), -1.0))
(
    1.0 * Hermitian(array([[-1., -0.],
       [-0., -1.]]), wires=[0])
  + -1.0 * X(0)
)
```

It updates the `data` attribute.  This is actually buggy for `Hamiltonian`, since we have:
```
H = 2.0 * qml.Hermitian(torch.eye(2), 0) + 3 * qml.X(0)

@qml.qnode(qml.device('lightning.qubit', wires=2))
def circuit():
    return qml.expval(H)

circuit()
```
```
AttributeError: 'Tensor' object has no attribute 'tobytes'
```

We aren't probably unwrapping all the data to numpy before sending it to the device.  This is something we need to fix.  I would be ok with fixing this in a "Bug Fix".

Reply:
Update, that error wasn't the result of not converting the data to numpy.... sorry.

Reply:
Is the conclusion to leave as is or remove registering LinearCombination to the same dispatch function?

Reply:
Oh yeah this needs to be resolved before approval as well.

Reply:
Leaving this with https://github.com/PennyLaneAI/pennylane/pull/5431 and the respective story. Not a blockers right now.